chapters/chapter_5/section_5.md

```diff
@@ -236,118 +263,62 @@ dummy_dd_new(GlobalConfig *cfg)
 }
+```
 
-extern CfgParser dummy_parser;
-
-static Plugin dummy_plugin =
-{
-  .type = LL_CONTEXT_DESTINATION,
-  .name = "dummy",
-  .parser = &dummy_parser,
-};
-
-gboolean
-dummy_module_init(PluginContext *context, CfgArgs *args)
-{
-  plugin_register(context, &dummy_plugin, 1);
+`DummyDriver` is `LogThreadedDestDriver`. A `LogThreadedDestDriver` has at least one `LogThreadedDestWorker`. It is the worker's responsibility to send the messages.
 
-  return TRUE;
-}
-
-const ModuleInfo module_info =
-{
-  .canonical_name = "dummy",
-  .version = SYSLOG_NG_VERSION,
-  .description = "This is a dummy destination for syslog-ng.",
-  .core_revision = SYSLOG_NG_SOURCE_REVISION,
-  .plugins = &dummy_plugin,
-  .plugins_len = 1,
-};
-```
+You can have as many workers as you need, but for simplest cases one worker is enough. In that case, for the sake of simplicity, you can implement `LogThreadedDestDriver` to work in a so called `compatibility mode`. This is how `DummyDriver` is implemented in the example above. `LogThreadedDestDriver` embeds a `LogThreadedDestWorker` instance. In compatibility mode, `LogThreadedDestDriver` will use the embedded worker. If `compatibility mode` is not enough for you, then you need to override the `construct` method of `LogThreadedDestDriver`.
 
 This unit can be separated into 5 parts:
 
 1. configuration functions (setters)
 2. utility functions
 3. functions running in separate thread
 4. functions running in the main thread
-5. plugin-construction declarations
 
 Functions from category 2-4 are implementations of the LogThreadedDestDriver's virtual methods.
 In `dummy_dd_new`, we pass these function pointers to the base classes.
 
 Our example overrides these 6 virtual methods:
 
-- `init (dummy_dd_init)`: destination constructor
-- `free_fn (dummy_dd_free)`: destination destructor
-- `thread_init (dummy_worker_thread_init)`: worker thread initialization
-- `thread_deinit (dummy_worker_thread_deinit)`: worker thread deinitialization
+- `new (dummy_dd_init)`: destination constructor.
+- `free_fn (dummy_dd_free)`: destination destructor.
+- `init (dummy_dd_init)`: It is called after startup, and after each reload. It is important to note that the init method may be called multiple times for the same driver. In case of a failed reload (for example syntax error in config), syslog-ng will resume using the same Driver instances instead of creating new ones, after calling their init method.
+- `deinit (dummy_dd_deinit)`: It is called before shutdown, and before each reload.
+- `thread_init (dummy_worker_thread_init)`: It is called after the init method.
+- `thread_deinit (dummy_worker_thread_deinit)`: It is called before the deinit method.
+- `connect (dummy_dd_connect)`: destination disconnects (on error or drop)
 - `disconnect (dummy_dd_disconnect)`: destination disconnects (on error or drop)
+- `insert (dummy_worker_insert)`, where you can format the received message and send to an actual destination.
```

Comment:
I agree that mentioning `flush` is not needed, as it is not mandatory.
Flush is useful for batching.

Even recently new threaded-dest drivers implemented first as having only the `insert` method (e.g. `kafka-c`).
If we don't mention batching, then flush is not necessary to describe.